### PR TITLE
support multiple app namespaces.

### PR DESCRIPTION
**Pull request by morvencao**
Context: support watching multiple namespaces for pilot

This is a followup PR for https://github.com/istio/istio/pull/13635

### CODE REVIEW

File: global.yaml [hunk 1]
```diff
@@ -21,6 +21,9 @@ global:
   istioNamespace: istio-control
   configNamespace: istio-control
 
+  # Applications namespace list pilot manages, separated by comma, if it contains multiple namespaces.
+  appNamespace: ""
```
Comment (sdake) [inline global.yaml]: shouldn't this be a pilot specific config in the pilot specific values.yaml file?

**Comment by costinm** [comment]
As in the other PR implementing this: 
- please not comma separate, use proper yaml ( i.e. a list )
- I'm not sure it's pilot only - mixer and all other components may need to follow the same list consistently. I suspect injector in particular. 

I'm not sure there is a use case for each component watching/using different namespaces.

**Comment by costinm** [comment]
You can still use the env variable in pilot for now - until we move this to mesh config and pilot gets it from there, but as API it should be exposed as a list, not comma separate string.  

We need to be very careful with all new fields and changes in values.yaml - and treat it as a highly visible API.

File: global.yaml (hunk 1 above)
Comment (costinm) [reply global.yaml]: +1, and also please no comma separate list in yaml !

**Comment by morvencao** [comment]
@costinm Not sure if this is OK, but if we want to use the env variable, then we need to transform the list to string.

**Comment by pinkertonpg** [comment]
The PR these changes were made for was never merged as far as I can tell: https://github.com/istio/istio/pull/13635

Should these changes be reverted?

**Pull request merged**